src/Fixer/Phpdoc/PhpdocNullableTypeDeclarationFixer.php

```diff
@@ -0,0 +1,237 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Phpdoc;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\DocBlock\Annotation;
+use PhpCsFixer\DocBlock\DocBlock;
+use PhpCsFixer\DocBlock\TypeExpression;
+use PhpCsFixer\Fixer\ConfigurableFixerInterface;
+use PhpCsFixer\Fixer\ConfigurableFixerTrait;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolver;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolverInterface;
+use PhpCsFixer\FixerConfiguration\FixerOptionBuilder;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\FixerDefinitionInterface;
+use PhpCsFixer\Preg;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @phpstan-type _AutogeneratedInputConfiguration array{
+ *  syntax?: 'question_mark'|'union',
+ * }
+ * @phpstan-type _AutogeneratedComputedConfiguration array{
+ *  syntax: 'question_mark'|'union',
+ * }
+ *
+ * @implements ConfigurableFixerInterface<_AutogeneratedInputConfiguration, _AutogeneratedComputedConfiguration>
+ *
+ * @no-named-arguments Parameter names are not covered by the backward compatibility promise.
+ */
+final class PhpdocNullableTypeDeclarationFixer extends AbstractFixer implements ConfigurableFixerInterface
+{
+    /** @use ConfigurableFixerTrait<_AutogeneratedInputConfiguration, _AutogeneratedComputedConfiguration> */
+    use ConfigurableFixerTrait;
+
+    private const OPTION_SYNTAX_UNION = 'union';
+    private const OPTION_SYNTAX_QUESTION_MARK = 'question_mark';
+
+    public function getDefinition(): FixerDefinitionInterface
+    {
+        return new FixerDefinition(
+            'Nullable PHPDoc type declarations should be standardised using the configured syntax.',
+            [
+                new CodeSample(
+                    <<<'PHP'
+                        <?php
+                        /** @var null|int $value */
+                        /** @param null|\Closure $callable */
+                        /** @return int|null */
+
+                        PHP
+                ),
+                new CodeSample(
+                    <<<'PHP'
+                        <?php
+                        /** @var ?int $value */
+                        /** @param ?\stdClass $obj */
+                        /** @return ?int */
+
+                        PHP,
+                    ['syntax' => self::OPTION_SYNTAX_UNION]
+                ),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
```

Comment:
at least `PhpdocTypesOrderFixer`, as new fixer can introduce `null|` that `PhpdocTypesOrderFixer` will move to last item